daemon/api.go

```diff
@@ -327,6 +332,22 @@ func getSnapsInfo(c *Command, r *http.Request) Response {
 		}
 	}
 
+	if len(includeTypes) > 0 {
+		for i, r := range results {
+			hasType := false
+			for _, t := range includeTypes {
+				if r["type"] == t {
+					hasType = true
+					break
+				}
+			}
+
+			if !hasType {
+				delete(results, i)
```

Comment:
Good point, having `i` as an integer index would definitely be the expected situation.